### PR TITLE
fixes

### PR DESCRIPTION
**Pull request by turuslan**
- update hunter, soralog, libp2p, filecoin-ffi, specs-actors.
- cgo `MinVerifiedDealSize`.
- fix `generateWinningPoStSectorChallenge` randomness.
- async `PaychGet`.
- (tmp) miner v5 state queries.
- fix mining sector selection.
- client verified storage deal.
- client storage async wait msg.
- client retrieval query and async `PaychGet`.
- soralog and spdlog to file.

### CODE REVIEW

File: core/api/full_node/make.cpp [hunk 1]
```diff
@@ -834,6 +866,22 @@ namespace fc::api {
       }
       return sectors;
     };
+    api->StateMinerAvailableBalance =
+        [=](auto &miner, auto &tsk) -> outcome::result<TokenAmount> {
+      OUTCOME_TRY(context, tipsetContext(tsk));
+      OUTCOME_TRY(actor, context.state_tree.get(miner));
+      OUTCOME_TRY(state, getCbor<MinerActorStatePtr>(context, actor.head));
+      // TODO(turuslan): older miner actor versions
```
Comment (wer1st) [inline core/api/full_node/make.cpp:874]: // TODO (m.tagirov)
I will do that in my task

File: core/api/full_node/node_api.hpp [hunk 2]
```diff
@@ -517,6 +522,10 @@ namespace fc::api {
                std::vector<SectorOnChainInfo>,
                const Address &,
                const TipsetKey &)
+    API_METHOD(StateMinerAvailableBalance,
+               TokenAmount,
```
Comment (wer1st) [inline core/api/full_node/node_api.hpp:526]:
```suggestion
               const TokenAmount &,
```

File: core/markets/retrieval/client/impl/retrieval_client_impl.hpp [hunk 3]
```diff
@@ -67,6 +67,9 @@ namespace fc::markets::retrieval::client {
      * Received ipld blocks verifier
      */
     Verifier verifier;
+
+    std::mutex pending_mutex;
```
Comment (wer1st) [inline core/markets/retrieval/client/impl/retrieval_client_impl.hpp:71]:
```suggestion
    mutable std::mutex pending_mutex;
```

File: core/vm/actor/builtin/v5/miner/state.hpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "vm/actor/builtin/states/miner/miner_actor_state.hpp"
+
+namespace fc::vm::actor::builtin::v5::miner {
```
Comment (wer1st) [inline core/vm/actor/builtin/v5/miner/state.hpp:10]: Add // TODO (m.tagirov)
Comment to remove this file

File: core/api/full_node/node_api.hpp (hunk 2 above)
Comment (turuslan) [reply core/api/full_node/node_api.hpp:526]: No, that's return type

Comment (wer1st) [reply core/api/full_node/node_api.hpp:526]: Oh, ok

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Review by wer1st — APPROVED** [review]

**Pull request merged**